### PR TITLE
Sende case validation

### PR DESCRIPTION
**Pull request by ihar-kahadouski**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/ws/annotations/In.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * limitations under the License.
+ */
+
+package com.epam.ta.reportportal.ws.annotations;
+
+import javax.validation.Constraint;
+import java.lang.annotation.*;
+
+/**
+ * Can be user with strings.
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/annotations/In.java]: used

File: src/main/java/com/epam/ta/reportportal/ws/annotations/InValidator.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+	@Override
+	public void initialize(In constraintAnnotation) {
+		allowedValues = constraintAnnotation.allowedValues();
+	}
+
+	@Override
+	public boolean isValid(String value, ConstraintValidatorContext context) {
+		if (null == value) {
+			return true;
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/annotations/InValidator.java]: Do we want to have 'null' allowed with this annotation? 
If yes - we need to describe the 'null' values behavior in documentation

File: src/main/java/com/epam/ta/reportportal/ws/model/StartRQ.java [hunk 3]
```diff
@@ -39,8 +38,7 @@
 @JsonInclude(Include.NON_NULL)
 public class StartRQ {
 
-	@NotNull
-	@NotEmpty
+	@NotBlank
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/model/StartRQ.java]: https://github.com/hibernate/hibernate-validator/blob/b173247ab36545b90f2bf64a3838f39f4aab2eb1/engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/NotBlankValidator.java#L33

Do we want to allow null values everywhere where NotEmpty and NotNull were?

File: src/main/java/com/epam/ta/reportportal/ws/model/StartTestItemRQ.java [hunk 4]
```diff
@@ -47,7 +44,7 @@
 	@JsonProperty(value = "uniqueId")
 	private String uniqueId;
 
-	@NotNull
+	@NotBlank
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/model/StartTestItemRQ.java]: https://github.com/hibernate/hibernate-validator/blob/b173247ab36545b90f2bf64a3838f39f4aab2eb1/engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/NotBlankValidator.java#L33

Do we want to allow null values everywhere where NotEmpty and NotNull were?

File: src/main/java/com/epam/ta/reportportal/ws/model/dashboard/CreateDashboardRQ.java [hunk 5]
```diff
@@ -33,20 +24,20 @@
...
  */
 @JsonInclude(Include.NON_NULL)
 @ApiModel
 public class CreateDashboardRQ extends SharableEntityRQ {
 
-	@NotNull
-	@NotEmpty
-	@Size(min = ValidationConstraints.MIN_NAME_LENGTH,
-	max = ValidationConstraints.MAX_DASHBOARD_NAME_LENGTH)
+	@NotBlank
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/model/dashboard/CreateDashboardRQ.java]: https://github.com/hibernate/hibernate-validator/blob/b173247ab36545b90f2bf64a3838f39f4aab2eb1/engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/NotBlankValidator.java#L33

Do we want to allow null values everywhere where NotEmpty and NotNull were?

File: src/main/java/com/epam/ta/reportportal/ws/model/StartRQ.java (hunk 3 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/model/StartRQ.java]: I checked with null and values. It works properly

File: src/main/java/com/epam/ta/reportportal/ws/model/StartTestItemRQ.java (hunk 4 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/model/StartTestItemRQ.java]: I checked with null and values. It works properly

File: src/main/java/com/epam/ta/reportportal/ws/model/dashboard/CreateDashboardRQ.java (hunk 5 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/model/dashboard/CreateDashboardRQ.java]: I checked with null and values. It works properly

File: src/main/java/com/epam/ta/reportportal/ws/annotations/In.java (hunk 1 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/annotations/In.java]: fixed

File: src/main/java/com/epam/ta/reportportal/ws/annotations/InValidator.java (hunk 2 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/annotations/InValidator.java]: Its already described in annotation docs

File: src/main/java/com/epam/ta/reportportal/ws/model/dashboard/CreateDashboardRQ.java (hunk 5 above)
Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/ws/model/dashboard/CreateDashboardRQ.java]: do you mean that null value is allowed in places where @NotNull was before?

Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/model/dashboard/CreateDashboardRQ.java]: I mean `@NotBlank` doesnt allow null and empty values. I checked it.

> The @NotNull class’ isValid() method is called after the @NotBlank class’ isValid(), hence forbidding null values.

Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/ws/model/dashboard/CreateDashboardRQ.java]: But the documentation of NotBlankValidator says different
https://github.com/hibernate/hibernate-validator/blob/b173247ab36545b90f2bf64a3838f39f4aab2eb1/engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/NotBlankValidator.java#L27

Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/model/dashboard/CreateDashboardRQ.java]: But documentation of annotation says that null - not valid. I suppose that one annotation can be handled by many validators in chain.
http://docs.jboss.org/hibernate/stable/validator/api/org/hibernate/validator/constraints/NotBlank.html

File: build.gradle [hunk 6]
```diff
@@ -37,7 +53,7 @@ dependencies {
     compile 'io.swagger:swagger-annotations:1.5.13'
     compile 'com.fasterxml.jackson.core:jackson-annotations'
     compile 'com.fasterxml.jackson.core:jackson-databind'
-    compile 'javax.validation:validation-api'
+    compile 'javax.validation:validation-api:2.0.1.Final'
```
Comment (pbortnik) [inline build.gradle]: we have https://github.com/reportportal/commons-bom
check it out and it s better probably good idea to move validation api dependency there

Comment (ihar-kahadouski) [reply build.gradle]: move versions in props

**Pull request merged**